test/metabase/api/search_test.clj

```diff
@@ -130,18 +138,20 @@
                         (merge (data-map instance-name)
                                (when-not in-root-collection?
                                  {:collection_id (u/the-id collection)})))]
-    (mt/with-temp* [Collection [coll         (data-map "collection %s collection")]
-                    Card       [action-model (if in-root-collection?
-                                               action-model-params
-                                               (assoc action-model-params :collection_id (u/the-id coll)))]
-                    Action     [action       (merge (data-map "action %s action")
-                                                    {:type :query, :model_id (u/the-id action-model)})]
-                    Card       [card         (coll-data-map "card %s card" coll)]
-                    Card       [dataset      (assoc (coll-data-map "dataset %s dataset" coll)
-                                                    :dataset true)]
-                    Dashboard  [dashboard    (coll-data-map "dashboard %s dashboard" coll)]
-                    Metric     [metric       (data-map "metric %s metric")]
-                    Segment    [segment      (data-map "segment %s segment")]]
+    (mt/with-temp* [Collection  [coll           (data-map "collection %s collection")]
+                    Card        [action-model   (if in-root-collection?
+                                                  action-model-params
+                                                  (assoc action-model-params :collection_id (u/the-id coll)))]
+                    Action      [{action-id :id
+                                  :as action}   (merge (data-map "action %s action")
+                                  {:type :query, :model_id (u/the-id action-model)})]
+                    QueryAction [_qa (query-action action-id)]
```

Comment:
And this is the test Action+QueryAction in the DB that will get found when searching.